### PR TITLE
#AJA

### PR DESCRIPTION
**Pull request by hideaki0817**
Introduce AJA ad network.

### CODE REVIEW

File: ads/_config.js [hunk 1]
```diff
@@ -216,6 +216,14 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
+  aja: {
+    renderStartImplemented: true,
+    prefetch: 'https://cdn.as.amanad.adtdp.com/sdk/asot-v2.js',
+    preconnect: [
```
Comment (calebcordry) [inline ads/_config.js]: You do not need to preconnect if url is in prefetch

File: ads/aja.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (calebcordry) [inline ads/aja.js]: nit: 2018

**Comment by hideaki0817** [comment]
- [ ] /ads/yournetwork.js: Implement the main logic here. This is the code that's invoked in the third-party iframe once loaded.
- [ ] /ads/yournetwork.md: Documentation detailing yourr ad service for publishers to read.
- [ ] /ads/_config.js: Add service specific configuration here.
- [ ] /3p/integration.js: Register your service here.
- [ ] /extensions/amp-ad/amp-ad.md: Add a link that points to your publisher doc.
- [ ] /examples/ads.amp.html: Add publisher examples here. Since a real ad isn't guaranteed to fill, a consistently displayed fake ad is highly recommended here to help AMP developers confidently identify new bugs.

File: ads/_config.js (hunk 1 above)
Comment (hideaki0817) [reply ads/_config.js]: thx. fixed it!

File: ads/aja.js (hunk 2 above)
Comment (hideaki0817) [reply ads/aja.js]: thx. fixed it!

**Comment by hideaki0817** [comment]
@calebcordry sorry to trouble you. do you mind checking again?

**Comment by hideaki0817** [comment]
@calebcordry sorry to trouble you. do you mind checking again?

File: ads/aja.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+export function aja(global, data) {
+
+  // ensure we have valid widgetIds value
+  validateData(data, ['widgetids']);
+
+  (global._aja = global._aja || {
+    viewId: global.context.pageViewId,
+    widgetIds: data['widgetids'],
+    htmlURL: data['htmlurl'] || global.context.canonicalUrl,
+    ampURL: data['ampurl'] || global.context.sourceUrl,
```
[the remark this answers is not in the document]
Comment (hideaki0817) [reply ads/aja.js]: thx I'll do that.

File: ads/aja.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import {loadScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function aja(global, data) {
+
+  // ensure we have valid widgetIds value
+  validateData(data, ['widgetids']);
```
[the remark this answers is not in the document]
Comment (hideaki0817) [reply ads/aja.js]: @calebcordry why do u think so? I think it doesn't work without widgetids.

**Comment by hideaki0817** [comment]
I just worked for ur all reviews:)

File: ads/aja.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (calebcordry) [reply ads/aja.js]: Sorry for the confusion, you are doing this correctly. You can also add the optional parameters as the *third* argument if you wish.

File: ads/aja.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+             layout="responsive"
+             data-widgetIds="AMP_1,AMP_2">
+  </amp-embed>
+```
+
+The above code must be accompanied by AMP-enabled widgets delivered by Aja’s Account Management Team, do not directly install this code with existing widgets.
+
+## Parameters
+
+- widgetIds *(**mandatory**)* - Widget Id/s Provided by Account Manager.
```
[the remark this answers is not in the document]
Comment (calebcordry) [reply ads/aja.md]: could you format this section like this to be consistent with other ad networks:

## Configuration

### Required Parameters

- `widgetIds` - Widget Id/s Provided by Account Manager.

### Optional parameters
- `htmlURL` - The URL of the standard html version of the page.
- `ampURL` - The URL of the AMP version of the page.
- `styleFile` - Provide publisher an option to pass CSS file in order to inherit the design for the AMP displayed widget. **Consult with Account Manager regarding CSS options**.

**Comment by calebcordry** [comment]
Also please remove the commited `yarn.lock` file.

**Comment by hideaki0817** [comment]
@calebcordry sorry.. How was that?

**Comment by calebcordry** [comment]
I am sorry, I don't understand the question. Do you need help removing the `yarn.lock` file?

**Comment by hideaki0817** [comment]
@calebcordry yes please if you don't mind. sorry..

**Comment by calebcordry** [comment]
No problem. Heres what I would do. 
1. `git reset --hard 865ff1ec79cb3cdaed6aafaad57ee15585e43fbb` This will reset your branch to the last commit before adding the `yarn.lock` file.
2. `git push origin master -f` this should update this pr

Let me know when you finish. Thanks

**Comment by hideaki0817** [comment]
image
@calebcordry I just tried it. but seems not changes. is it correct now?

**Comment by calebcordry** [comment]
I can try on my end. Can you make sure that `allow edits from maintainers` is checked.

image

**Comment by calebcordry** [comment]
Oops. Sorry I messed up this branch trying to help. Do you mind opening a new PR and I will approve. Thanks.

**Comment by hideaki0817** [comment]
thank you. no problem. I just created PR on https://github.com/ampproject/amphtml/pull/16306

**Pull request closed without merging**